relay-system/src/compat.rs

```diff
@@ -57,3 +57,26 @@ where
     EXECUTOR.get().unwrap().spawn(f);
     rx.await.map_err(|_| MailboxError::Closed)?
 }
+
+/// Sends a message to a recipient using `actix` 0.7 from a `tokio` 1.0 context.
+///
+/// The message is internally forwarded through a `tokio` 0.1 runtime in order to avoid panics when
+/// the channel queues fill up and tasks are getting parked.
+///
+/// # Panics
+///
+/// Panics if this is invoked outside of the [`Controller`](crate::Controller).
+///
+/// NOTE: required by ProjectCache and will be removed with ProjectCache migration.
+pub async fn send_to_recipient<M>(addr: Recipient<M>, msg: M) -> Result<M::Result, MailboxError>
+where
+    M: Message + Send + 'static,
+    M::Result: Send,
+{
+    let (tx, rx) = oneshot::channel();
+    let f = futures01::future::lazy(move || addr.send(msg))
+        .then(|res| tx.send(res))
+        .map_err(|_| ());
+    EXECUTOR.get().unwrap().spawn(f);
+    rx.await.map_err(|_| MailboxError::Closed)?
```

Comment:
Of what type is the error that we map to `MailboxError::Closed` here?

Reply:
The type of error is `tokio::sync::oneshot::error::RecvError` from the receiver's [`Future` implementation](https://docs.rs/tokio/latest/tokio/sync/oneshot/struct.Receiver.html#associatedtype.Output). Semantics are compatible for this case.